File: ts/src/data/point.ts
export interface Point2D {
  x: number;
  y: number;
}

export interface Point3D {
  x: number;
  y: number;
  z: number;
}

export function getDistanceBetween(
  p1: Point2D,
  p2: Point2D,
) {
  const deltaX = p2.x - p1.x;
  const deltaY = p2.y - p1.y;
  return Math.sqrt(Math.pow(deltaX, 2) + Math.pow(deltaY, 2));
}

export function getDegreeBetween(
  p1: Point2D,
  p2: Point2D,
) {
  const deltaX = p2.x - p1.x;
  const deltaY = p2.y - p1.y;
  return radianToDegree(Math.atan2(deltaY, deltaX));
}

export function degreeToRadian(degree: number) {
  return (degree * Math.PI) / 180;
}

export function radianToDegree(radian: number) {
  return (radian * 180) / Math.PI;
}

File: python/src/data/point.py
from typing import TypedDict
import math

class Point2D(TypedDict):
    x: float
    y: float

class Point3D(Point2D):
    z: float

def getDistanceBetween(p1: Point2D, p2: Point2D) -> float:
    deltaX = p2.x - p1.x
    deltaY = p2.y - p1.y
    return math.sqrt(deltaX ** 2 + deltaY ** 2)

def getDegreeBetween(p1: Point2D, p2: Point2D) -> float:
    deltaX = p2.x - p1.x
    deltaY = p2.y - p1.y
    return radianToDegree(math.atan2(deltaY, deltaX))

def degreeToRadian(degree: float) -> float:
    return (degree * math.pi) / 180

def radianToDegree(radian: float) -> float:
    return (radian * 180) / math.pi